File: 2121-find-if-path-exists-in-graph/find-if-path-exists-in-graph.java
class Solution {
    public boolean validPath(int n, int[][] edges, int source, int destination) {
        Map<Integer,List<Integer>>graph = new HashMap<>();
        for(int []edge:edges)
        {
            int u =edge[0];
            int v = edge[1];
            graph.computeIfAbsent(u, val->new ArrayList<>()).add(v);
            graph.computeIfAbsent(v,val -> new ArrayList<>()).add(u);

        }
            
            Queue<Integer>queue = new LinkedList<>();
            boolean [] seen = new boolean[n];
            seen[source] =true;

            queue.offer(source);

            while(!queue.isEmpty())
            {
                int currNode = queue.poll();
                if(currNode == destination)
                {
                    return true;
                }

                for(int nextNode : graph.get(currNode))
                {
                    if(!seen[nextNode])
                    {
                        seen[nextNode] = true;
                        queue.offer(nextNode);
                    }
                }
            }
      return false;
        
    }
}

File: 2121-find-if-path-exists-in-graph/find-if-path-exists-in-graph.cpp
class Solution {
public:
    bool validPath(int n, vector<vector<int>>& edges, int source, int destination) {
        unordered_map<int,vector<int>>graph;
        for(auto & edge:edges)
        {
            int a=edge[0];
            int b=edge[1];
            graph[a].push_back(b);
            graph[b].push_back(a);

        }
        vector<bool>seen(n);
        seen[source]=true;
        queue<int>bfs;
        bfs.push(source);
        while(!bfs.empty())
        {
            int currNode=bfs.front();
            seen[currNode]=true;
            bfs.pop();

            if(currNode == destination)
            {
                return true;
            }
            for(auto nextnode: graph[currNode])
            {
                if(!seen[nextnode])
                {
                    seen[nextnode]=true;
                    bfs.push(nextnode);
                }
            }
        }

return false;

    }
};